packages/cpp/ArmoniK.Api.Client/source/submitter/SubmitterClient.cpp

```diff
@@ -0,0 +1,452 @@
+  for (const auto& partition_id : partition_ids)
+  {
+    request.add_partition_ids(partition_id);
+  }
+  CreateSessionReply reply;
+  
+  Status status = stub_->CreateSession(&context_, request, &reply);
+  if (!status.ok())
+  {
+    std::stringstream message;
+    message << "Error: " << status.error_code() << ": " << status.error_message()
+            << ". details : " << status.error_details() << std::endl;
+    std::cout << "CreateSession rpc failed: " << std::endl;
+    throw std::runtime_error(message.str().c_str());
+  }
+  return reply.session_id();
+}
+
+/**
+ * @brief Cancel a session.
+ *
+ * @param session_id The session id.
+ */
+bool SubmitterClient::cancel_session(std::string_view session_id)
+{
+  std::shared_ptr<Channel> channel = grpc::CreateChannel(
+      "172.30.39.223:5001",
+                                grpc::InsecureChannelCredentials());
+  auto stub = armonik::api::grpc::v1::sessions::Sessions::NewStub(channel);
+  armonik::api::grpc::v1::sessions::CancelSessionRequest request;
+
+  request.set_session_id(static_cast<std::string>(session_id));
+
+  armonik::api::grpc::v1::sessions::CancelSessionResponse reply;
+  grpc::ClientContext context;
+
+  Status status = stub->CancelSession(&context, request, &reply);
+
+  if (!status.ok())
+  {
+    std::stringstream message;
+    message << "Error: " << status.error_code() << ": " << status.error_message()
+            << ". details : " << status.error_details() << std::endl;
+    std::cout << "CreateSession rpc failed: " << std::endl;
+    throw std::runtime_error(message.str().c_str());
+  }
+  
+  return true;
+}
+
+
+/**
+ * @brief Convert task_requests to request_stream.
+ *
+ * @param task_requests A vector of TaskRequest objects.
+ * @param session_id The session ID.
+ * @param task_options The TaskOptions object.
+ * @param chunk_max_size Maximum chunk size.
+ * @return A vector of futures containing CreateLargeTaskRequest objects.
+ */
+std::vector<std::future<std::vector<
+  CreateLargeTaskRequest>>> SubmitterClient::to_request_stream(const std::vector<TaskRequest>& task_requests,
+                                                                  std::string& session_id,
+                                                                  const TaskOptions& task_options,
+                                                                  const size_t chunk_max_size)
+{
+  std::vector<std::future<std::vector<CreateLargeTaskRequest>>> async_chunk_payload_tasks;
+  async_chunk_payload_tasks.push_back(std::async([session_id, task_options]()
+  {
+    CreateLargeTaskRequest_InitRequest create_large_task_request_init;
+    create_large_task_request_init.set_session_id(session_id);
+    *create_large_task_request_init.mutable_task_options() = task_options;
+
+    CreateLargeTaskRequest create_large_task_request;
+    *create_large_task_request.mutable_init_request() = create_large_task_request_init;
+
+    return std::vector{create_large_task_request};
+  }));
```

Comment:
Once you take `session_id` and `task_options` by value, you can move out those values into the lambda:
```suggestion
  CreateLargeTaskRequest>>> SubmitterClient::to_request_stream(const std::vector<TaskRequest>& task_requests,
                                                                  std::string session_id,
                                                                  TaskOptions task_options,
                                                                  const size_t chunk_max_size)
{
  std::vector<std::future<std::vector<CreateLargeTaskRequest>>> async_chunk_payload_tasks;
  async_chunk_payload_tasks.push_back(std::async([session_id = std::move(session_id), task_options = std::move(task_options)]()
  {
    CreateLargeTaskRequest_InitRequest create_large_task_request_init;
    create_large_task_request_init.set_session_id(std::move(session_id));
    create_large_task_request_init.set_task_options(std::move(task_options));

    CreateLargeTaskRequest create_large_task_request;
    create_large_task_request.set_init_request(std::move(create_large_task_request_init));

    return std::vector{create_large_task_request};
  }));
```